packages/meteor/src/MethodDataFetcherImpl.ts

```diff
@@ -112,7 +112,7 @@ export abstract class MethodDataFetcherImpl<Q, D> implements MeteorDataFetcher<Q
             throw new Error('setDataFetcher was not called!');
         }
         this.cancelCurrentRequest();
-        this._dataReady = false;
+        action(() => (this._dataReady = false));
```

Comment:
I'm not familiar with the implications of creating an action on the fly.
Can't we  put the `@action` decorator around the whole `runDataFetcherFunction()` method?

BTW, I think with the current code, we are not even executing what we want to execute, because the `action()` function we are wrapping around it just returns a wrapped function, which we are not calling. 

If we want to wrap it on the fly _and_ call it (which I'm not sure we want), when I think we need `action(() => { whatever })()`
 (notice the `()` at the end.)